### ISSUE TITLE
`_get_size_of_locals` fails for non-hashable values

### ISSUE THREAD

**Issue by drewler** [jg-rp/liquid#79]
When setting resource limits for a `Context`'s by setting a `local_namespace_limit`, it fails when computing `_get_size_of_locals` if any of the values is not hashable (eg: a `list`) because it tries to de-duplicate them by putting them in a `set`.

https://github.com/jg-rp/liquid/blob/e123ce078bdda3963a3589a98089792a06163d0e/liquid/context.py#L468-L473

Code reproducing the issue:

```python
from liquid import Environment

class ResourceLimitedEnvironment(Environment):
    # Limits the number of variables that can be defined within a template.
    local_namespace_limit = 100

template_string = """
{% assign beatles = "John, Paul, George, Ringo" | split: ", " %}

{% for member in beatles %}
  {{ member }}
{% endfor %}
"""

# Works
default_env = Environment()
template = default_env.from_string(template_string)
template.render()

# Doesn't work
resource_limited_env = ResourceLimitedEnvironment()
resource_limited_template = resource_limited_env.from_string(template_string)
resource_limited_template.render()
```

**Comment by jg-rp** [comment]
Hi @drewler,

Thank you for reporting this bug.

Pull request #80 fixes this issue by not not building a `set` from a render context's local namespace. In the unlikely event that someone is relying on the old, broken behaviour, `Context` now exposes `get_size_of_locals()` for potential customization of the local namespace "size" calculation.

This example mimics the broken behaviour.

```python
import sys

from liquid import Environment
from liquid import Context
from liquid.template import BoundTemplate

class MyRenderContext(Context):
    def get_size_of_locals(self) -> int:
        if not self.env.local_namespace_limit:
            return 0
        return (
            sum(sys.getsizeof(obj, default=1) for obj in set(self.locals.values()))
            + self.local_namespace_size_carry
        )


class MyBoundTemplate(BoundTemplate):
    context_class = MyRenderContext


class MyEnvironment(Environment):
    local_namespace_limit = 110  # XXX: very low, for demonsrtation purposes
    template_class = MyBoundTemplate


env = MyEnvironment()

template = env.from_string(
    "{% assign foo = 'hello' %}"
    "{% assign bar = 'world' %}"
).render()


# raises a LocalNamespaceLimitError
template = env.from_string(
    "{% assign foo = 'hello' %}"
    "{% assign bar = 'world' %}"
    "{% assign baz = '!' %}"
).render()
```

And this example simply counts the number of items in the namespace.

```python
from liquid import Environment
from liquid import Context
from liquid.template import BoundTemplate

class MyRenderContext(Context):
    def get_size_of_locals(self) -> int:
        if not self.env.local_namespace_limit:
            return 0
        return len(self.locals) + self.local_namespace_size_carry


class MyBoundTemplate(BoundTemplate):
    context_class = MyRenderContext


class MyEnvironment(Environment):
    local_namespace_limit = 2  # XXX: very low, for demonsrtation purposes
    template_class = MyBoundTemplate


env = MyEnvironment()

template = env.from_string(
    "{% assign foo = 'hello' %}"
    "{% assign bar = 'world' %}"
).render()

# raises a LocalNamespaceLimitError
template = env.from_string(
    "{% assign foo = 'hello' %}"
    "{% assign bar = 'world' %}"
    "{% assign baz = '!' %}"
).render()
```

**Comment by jg-rp** [comment]
Fixed in version 1.4.7. Now released.

**Comment by drewler** [comment]
That was fast. Thank you @jg-rp!
